src/FakeItEasy/Configuration/FakeConfigurationManager.cs

```diff
@@ -100,12 +102,14 @@ private static Expression BuildArgumentThatMatchesAnything<TValue>()
 
         private static FakeManager GetFakeManagerCallIsMadeOn(ParsedCallExpression parsedCallExpression)
         {
-            if (parsedCallExpression.CallTarget == null)
-            {
-                throw new ArgumentException("The specified call is not made on a fake object.");
-            }
+            return parsedCallExpression.CallTarget == null
+                ? null

```

Comment:
Well spotted. I think I like it. You're okay relying on `FakeFacadeManager.GetFakeManager` complaining that the faked object is null, if ever it comes up, not that I think it will…

About inlining `Fake.GetFakeManager`, do you mean everywhere? If we remove `EnsureCallIsMadeOnAFake`, `CallTo(Expression<Action>)` becomes

```c#
public IVoidArgumentValidationConfiguration CallTo(Expression<Action> callSpecification)
{
    Guard.AgainstNull(callSpecification, nameof(callSpecification));

    var parsedCallExpression = this.callExpressionParser.Parse(callSpecification);
    Fake.GetFakeManager(parsedCallExpression.CallTarget);
    this.AssertThatMemberCanBeIntercepted(parsedCallExpression);

    return this.CreateVoidArgumentValidationConfiguration(parsedCallExpression);
}
```

In this case, there's an unused return value from `Fake.GetManager` which looks funny. (There's _always_ an unused return value, but if it's hidden in `EnsureCallIsMadeOnAFake`, then it's not so weird looking.)